client/src/modules/email-report/email-report.ctrl.js

```diff
@@ -0,0 +1,230 @@
+angular.module('bhima.controllers')
+.controller('EmailReportController', EmailReportController);
+
+EmailReportController.$inject = [
+  '$state', 'EmailReportService',
+  'ReportGroupService', 'SessionService', 'util',
+  'NotifyService', 'ScrollService', 'bhConstants', 'uiGridConstants',
+];
+
+/**
+ * Report group Controller
+ *
+ * @description
+ *this controller interact with the view in order to manage
+ * people's profiles (emails and name)
+ * The report is automaticly generated by bhima and  will be sent throw those emails
+ */
+
+function EmailReportController($state, EmailReportSvc, ReportGroupSvc
+  , Session, util, Notify, ScrollTo, bhConstants, uiGridConstants) {
+
+  var vm = this;
+  /*
+   emailReport is the object
+   that contains : name,email, frequence and report-group code in UI form
+  */
+  vm.emailReport = {};
+  vm.selectedEmailReport = { selected : false };
+  vm.save = save;
+  vm.remove = remove;
+  vm.frequencies = [];
+  /*
+    when saving the profile, the user should select a report group
+    in orther to specify what report this email will get
+    the bellow array will contains those report
+  */
+  vm.report_groups = [];
+
+  init();
+
+  /*
+   initialise importants values for the UI form
+   such as the frequencies and report groups
+   that should be loaded from ReportGroupService
+
+   */
+
+
+  function init() {
+    /* loading frequencies*/
+    vm.frequencies = EmailReportSvc.frequencies;
+
+    /* If the user has selected a profile, fill name and email field */
+    try {
```

Comment:
If you default `data` to an empty object, you will not need this `try/catch` block.